src/framework/tf_framework.py

```diff
@@ -72,23 +74,23 @@ def checkpoint(self, epoch, step_number):
         """
         if self.rank() == 0:
             my_rank = self.rank()
-            if not os.path.exists(self.checkpoint_folder):
-                os.makedirs(self.checkpoint_folder)
+            if not self.storage.exists(self.checkpoint_folder):
+                self.storage.makedirs(self.checkpoint_folder)
 
-            model_file = os.path.join(self.checkpoint_folder, f"model-{epoch}-{step_number}.bin")
-            meta_file = os.path.join(self.checkpoint_folder, f"meta-{epoch}-{step_number}.bin")
-            index_file = os.path.join(self.checkpoint_folder, f"index-{epoch}-{step_number}.bin")
+            model_file = self.storage.get_uri(os.path.join(self.checkpoint_folder, f"model-{epoch}-{step_number}.bin"))
+            meta_file = self.storage.get_uri(os.path.join(self.checkpoint_folder, f"meta-{epoch}-{step_number}.bin"))
+            index_file = self.storage.get_uri(os.path.join(self.checkpoint_folder, f"index-{epoch}-{step_number}.bin"))
 
-            f = open(model_file, "w")
+            f = tf.io.gfile.GFile(model_file, "w")
```

Comment:
Even writing checkpoints should be abstracted through the storage interface. As in the absence of a filesystem we would have to write the checkpoints on S3 again.